search/includes/classes/class-health.php

```diff
@@ -322,10 +335,26 @@ public static function validate_index_posts_content( $start_post_id = 1, $last_p
 			}
 		} while ( $start_post_id <= $last_post_id );
 
+		$this->delete_validate_content_ongoing();
+
 		return $results;
 	}
 
-	public static function validate_index_posts_content_batch( $indexable, $start_post_id, $next_batch_post_id, $inspect ) {
+	public function is_validate_content_ongoing(): bool {
+		$is_locked = get_transient( self::CONTENT_VALIDATION_LOCK_NAME, false );
+
+		return (bool) $is_locked;
+	}
+
+	public function reset_validate_content_ongoing() {
+		set_transient( self::CONTENT_VALIDATION_LOCK_NAME, true, self::CONTENT_VALIDATION_LOCK_TIMEOUT );
+	}
+
+	public function delete_validate_content_ongoing() {
```

Comment:
I think this should be `remove_validate_content_lock` or `delete_validate_content_lock`. The reasoning is we're operating on a lock.

What do you think?